App/ViewModels/FileSyncListViewModel.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Windows.Storage.Pickers;
+using Coder.Desktop.App.Models;
+using Coder.Desktop.App.Services;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Microsoft.UI.Dispatching;
+using Microsoft.UI.Xaml;
+using WinRT.Interop;
+
+namespace Coder.Desktop.App.ViewModels;
+
+public partial class FileSyncListViewModel : ObservableObject
+{
+    public delegate void OnFileSyncListStaleDelegate();
+
+    // Triggered when the window should be closed.
+    public event OnFileSyncListStaleDelegate? OnFileSyncListStale;
```

Comment:
I think it's confusing UX to just close out the window if something is wrong. It'll seem like a bug if you're creating a new sync session and the window just disappears.

If you end up creating a sync session that doesn't work for some reason, we should just show it in an error state.

Reply:
Users will not be able to open the window if the VPN is not active. This handles closing the window when the VPN session is closed.

Reply:
Preventing them from opening it is fine if it's reasonably clear why, as is closing it if it is in response to some _action_ the user has taken that can be reasonably seen as related.

It looks to me like we would close the window if the VPN service crashed or otherwise went down other than via user action.

Reply:
That behavior sounds fine to me. The user won't be able to add new sync sessions since we won't have a list of agents to show in the list, so the UI will already have reduced functionality anyways

Reply:
It's not about what the user can and can't reasonably do---it's about whether they can understand why things are happening. Think about it from the user's perspective: if you're setting up a file sync session and the window just suddenly closes, what do you think? Probably that the app crashed; that it's glichy and buggy and not to be trusted.

Reply:
Would you prefer if the window had a "unavailable" state?

Reply:
Just having it would be an improvement, since it would look less like things crashed. If it explained why its unavailable that would be even better.

Reply:
Added some text that shows when file syncs aren't available based on the RpcLifecycle, VpnLifecycle or CredentialState